dali/operators/math/expressions/arithmetic_test.cc

```diff
@@ -92,12 +94,35 @@ TEST(ArithmeticOpsTest, PreservePseudoScalarInput) {
   }
   ws.AddInput(in[0]);
 
-  auto result_shape = PropagateShapes<CPUBackend>(expr_ref, ws, 2);
-  auto expected_shpe = TensorListShape<>{{1}, {1}};
-  EXPECT_EQ(result_shape, expected_shpe);
+  TensorListShape<> result_shape;
+  PropagateShapes<CPUBackend>(result_shape, expr_ref, ws, 2);
+  auto expected_shape = TensorListShape<>{{1}, {1}};
+  EXPECT_EQ(result_shape, expected_shape);
 }
 
-TEST(ArithmeticOpsTest, TreePropagationError) {
+TEST(ArithmeticOpsTest, TreePropagationBroadcasting) {
+  std::string expr_str = "div(sub(&0 &1) &2)";
+  auto expr = ParseExpressionString(expr_str);
+  auto &expr_ref = *expr;
+  HostWorkspace ws;
+  std::shared_ptr<TensorList<CPUBackend>> in[3];
+  for (auto &ptr : in) {
+    ptr = std::make_shared<TensorList<CPUBackend>>();
+    ptr->Resize({{1}, {2}}, DALI_INT32);
+  }
+  in[2]->Resize({{10}, {2}});
+  ws.AddInput(in[0]);
+  ws.AddInput(in[1]);
+  ws.AddInput(in[2]);
+
+  TensorListShape<> result_shape;
+  ASSERT_TRUE(PropagateShapes<CPUBackend>(result_shape, expr_ref, ws, 2));
+  TensorListShape<> expected_sh = {{10}, {2}};
+  ASSERT_EQ(expected_sh, result_shape);
```

Comment:
Nitpick: use EXPECT rather than ASSERT when there's no danger of causing a hard error or flooding the output.